tests/unit/net/fetch.rs

```diff
@@ -506,6 +512,55 @@ fn test_fetch_with_local_urls_only() {
     assert!(server_response.is_network_error());
 }
 
+#[test]
+fn test_fetch_with_hsts() {
+    static MESSAGE: &'static [u8] = b"";
+    let handler = move |_: HyperRequest, response: HyperResponse| {
+        response.send(MESSAGE).unwrap();
```

Comment:
If I correctly understand, you want `request.url.scheme()` to be sent as part of response body?. Here we use `hyper::server::request::Request` which does not capture information about request scheme. If we make a non-HTTPS connection to the server, then we get response error saying "Connection reset"